src/libYARP_OS/src/Node.cpp

```diff
@@ -350,6 +436,10 @@ void yarp::os::Node::Helper::add(Contactable& contactable)
     }
     prepare(name);
     item.contactable = &contactable;
+    ROSReport* rep = new ROSReport();
+    item.contactable->setReporter(*rep);
```

Comment:
I am not sure .. However I think it's not possible.
The contactable is stored in `name_cache` of `yarp::os::Node::Helper`, which is a public attribute. However, the helper is owned by an instance of  `yarp::os::Node`, which does not allow access to the helper, and thus not to the contactable either. So the only places to access the contactable are within the helper class and the `Node` class.